packages/api-fetch/src/index.js

```diff
@@ -78,9 +83,22 @@ const defaultFetchHandler = ( nextOptions ) => {
 	);
 
 	return responsePromise
-		.then( checkStatus )
-		.catch( ( response ) => parseAndThrowError( response, parse ) )
-		.then( ( response ) => parseResponseAndNormalizeError( response, parse ) );
+		// Return early if fetch errors. If fetch error, there is most likely no
+		// network connection. Unfortunately fetch just throws a TypeError and
+		// the message might depend on the browser.
+		.then(
+			( value ) =>
+				Promise.resolve( value )
+					.then( checkStatus )
+					.catch( ( response ) => parseAndThrowError( response, parse ) )
+					.then( ( response ) => parseResponseAndNormalizeError( response, parse ) ),
+			() => {
+				throw {
+					code: 'fetch_error',
+					message: __( 'You are probably offline.' ),
+				};
+			}
+		);
```

Comment:
I wonder if we can unit test that behavior.

Reply:
Good point. Added one.